properties/base/containers.py

```diff
@@ -459,3 +471,175 @@ def from_json(value, **kwargs):
         set's prop type is unknown.
         """
         return set(value)
+
+
+class Dictionary(basic.Property):
+
+    class_info = 'a dictionary'
+    _class_default = dict
+
+    @property
+    def observe_mutations(self):
+        """observe_mutations makes all mutations fire change notifications"""
+        return getattr(self, '_observe_mutations', False)
+
+    @observe_mutations.setter
+    def observe_mutations(self, value):
+        if not isinstance(value, bool):
+            raise TypeError('observe_mutations must be a boolean')
+        self._observe_mutations = value
+
+    @property
+    def key_prop(self):
+        """Property type allowed for keys"""
+        return getattr(self, '_key_prop', basic.Property(''))
+
+    @key_prop.setter
+    def key_prop(self, value):
+        self._key_prop = validate_prop(value)
+
+    @property
+    def value_prop(self):
+        """Property type allowed for values"""
+        return getattr(self, '_value_prop', basic.Property(''))
+
+    @value_prop.setter
+    def value_prop(self, value):
+        self._value_prop = validate_prop(value)
+
+    @property
+    def name(self):
+        return getattr(self, '_name', '')
+
+    @name.setter
+    def name(self, value):
+        if self.key_prop:
+            self.key_prop.name = value
+        if self.value_prop:
+            self.value_prop.name = value
+        self._name = value
+
+    @property
+    def info(self):
+        """Supplemental description of the list, with length and type"""
+        itext = self.class_info
+        if self.key_prop.info and self.value_prop.info:
+            itext += ' (keys: {}; values: {})'.format(
+                self.key_prop.info, self.value_prop.info
+            )
+        elif self.key_prop.info:
+            itext += ' (keys: {})'.format(self.key_prop.info)
+        elif self.value_prop.info:
+            itext += ' (values: {})'.format(self.value_prop.info)
+        return itext
+
+    def validate(self, instance, value):
+        if not isinstance(value, dict):
```

Comment:
I think maybe this is a bit too strict. At least, it would be nice to allow for `OrderedDict` without coercion. Thoughts?

Reply:
Even something like `allow_types=[..., ...]` to specify classes allowed to pass this.

The issue here is that it's not duck-typing friendly, and Python has a whole lot of ducks that look like dictionaries...